### PR TITLE
Add ui tests

### PR DESCRIPTION
**Pull request by fcollonval**
Add playwright to test the examples

- Remaining examples:
  - [x] settings
  - [x] state
  - [x] toolbar-button
  - [x] widget
- [x] Tests on CI

This uncovers a bug in style packaging.  
This is using playwright test runner (no galata).

There are nice examples of playwright code for:
- capturing log console message (hello-world, settings,...), 
- capturing alert message (settings,...), 
- emulate drag and drop (move a panel in the main area to create a split panel) (kernel-output & log-messages).
- get `select` current value (state)
- wait for a request (state)

log console and alert dialogs are not capture when generating videos from tests. So it seems unappropriate to replace existing preview with those videos although they will allow to be up to date.

What do you think @jtpio @echarles?

### CODE REVIEW

**Comment by jtpio** [comment]
Thanks for starting this!

>  This is using playwright test runner (no galata).

This could be useful to lay the foundations for a potential rework in Galata discussed in https://github.com/jupyterlab/galata/issues/67

> So it seems unappropriate to replace existing preview with those videos although they will allow to be up to date.

Some time a while ago it was also discussed whether to check the screencasts in the repositories because of their size. Probably this will be a similar issue with videos?

**Comment by fcollonval** [comment]
> Probably this will be a similar issue with videos?

Yes. What I had in mind was to create a manual workflow (or one triggered by new JLab release) that would have open a PR to update those videos/screenshots only once in a while (with maybe a conversion to GIF and cutting the first few seconds corresponding to the logo animation for the videos).

**Comment by fcollonval** [comment]
The failure is on Mac and not related to this.

File: command-palette/ui-tests/README.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
+# Test
```
Comment (jtpio) [inline command-palette/ui-tests/README.md:1]: Maybe we should also diff these files on CI to make sure they stay in sync?

- `ui-tests/README.md`
- `ui-tests/playwright.config.ts`

Comment (fcollonval) [reply command-palette/ui-tests/README.md:1]: Done thanks for the suggestion
> Note I added test for the server-extension too.

**Comment by jtpio** [comment]
Probably this could fix https://github.com/jupyterlab/extension-examples/issues/45?

Or do we still want to have examples of setting up unit tests in this repo?

**Review by jtpio — APPROVED** [review]
Looks good, thanks!

**Comment by jtpio** [comment]
Also we might want to make the `ui-tests` more discoverable, for example by linking to the folder from the example README.md?

So folks interesting in this could learn how to set them up. This could be in the hello world example, or in a dedicated testing example.

**Comment by fcollonval** [comment]
I would rather create a dedicated example to talk about testing (addressing ui-tests and jest tests). But I want to work on galata to stabilize its usage with playwright before doing so.

I'll merge this one and open a PR to add a section about the tests in the README.

**Pull request merged**